lib/application.js

```diff
@@ -198,8 +198,9 @@ class Application extends EggApplication {
 
   handleRequest(ctx, fnMiddleware) {
     this.emit('request', ctx);
-    super.handleRequest(ctx, fnMiddleware);
+    const promise = super.handleRequest(ctx, fnMiddleware);
     onFinished(ctx.res, () => this.emit('response', ctx));
```

Comment:
```js
onFinished(ctx.res, () => this.emit('response', ctx));
return super.handleRequest(ctx, fnMiddleware);
```

please add a test case for this change to ensure it won't be broken in the future.

Reply:
Got it, I am done.